services/vehicle/src/main/java/com/wedriveu/services/vehicle/entity/VehicleStoreImpl.java

```diff
@@ -39,64 +41,171 @@ public void mapEntityToJson() {
+                                  Position position,
+                                  Date lastUpdate) {
+        return new Vehicle(carLicencePlate, state, position, lastUpdate);
     }
 
     private void writeJsonVehicleFile(ArrayList<Vehicle> vehicleListToJSon) {
         ObjectMapper mapper = new ObjectMapper();
-
-        try {
-            mapper.writeValue(new File(Constants.VEHICLES_DATABASE_PATH), vehicleListToJSon);
-            String jsonInString = mapper.writeValueAsString(vehicleListToJSon);
-            Log.log(jsonInString);
-            jsonInString = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(vehicleListToJSon);
-            Log.log(jsonInString);
-
-        } catch (JsonGenerationException e) {
-            e.printStackTrace();
-        } catch (JsonMappingException e) {
-            e.printStackTrace();
-        } catch (IOException e) {
-            e.printStackTrace();
-        }
+        checkDuplicatesAndWriteOnVehiclesDb(vehicleListToJSon, mapper);
     }
 
-    private Vehicle getVehicleRequestedCheckingVehiclesList(Vehicle[] vehicles, String carLicencePlate) {
+    private Vehicle getVehicleRequestedCheckingVehiclesList(List<Vehicle> vehicles, String carLicencePlate) {
         for (Vehicle vehicle : vehicles) {
             if(vehicle.getCarLicencePlate().equals(carLicencePlate)){
-                Log.log("com.wedriveu.services.vehicle.entity.Vehicle found! -> " + vehicle.getCarLicencePlate() + " " + vehicle.getState());
+                Log.log("com.wedriveu.services.vehicle.entity.Vehicle found! -> " +
+                        vehicle.getCarLicencePlate() +
+                        " " +
+                        vehicle.getState());
                 return vehicle;
             }
         }
         Log.log("com.wedriveu.services.vehicle.entity.Vehicle not found, retry!");
         return null;
     }
 
+    private List<Vehicle> readFromVehiclesDb(ObjectMapper mapper){
+        try {
+            List<Vehicle> vehicles =
+                    mapper.readValue(new File(Constants.VEHICLES_DATABASE_PATH), new TypeReference<List<Vehicle>>(){});
+            return vehicles;
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    private void checkDuplicatesAndWriteOnVehiclesDb(List<Vehicle> vehicles, ObjectMapper mapper) {
+        if(thereAreNotDuplicates(vehicles)) {
+            try {
+                mapper.writeValue(new File(Constants.VEHICLES_DATABASE_PATH), vehicles);
+                String jsonInString = mapper.writeValueAsString(vehicles);
+                Log.log(jsonInString);
+                jsonInString = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(vehicles);
+                Log.log(jsonInString);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    private boolean thereAreNotDuplicates(List<Vehicle> vehicles) {
```

Comment:
Grammatically is more correct like this, but i changed it, as you prefer :) 
I'm ok in any case.